### PR TITLE
feature : add efCore queryable extensions

### PR DESCRIPTION
**Pull request by wzh425**

### CODE REVIEW

File: src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/_Imports.cs [hunk 1]
```diff
@@ -16,3 +16,4 @@
 global using Microsoft.Extensions.Options;
 global using System.Linq.Expressions;
 global using System.Reflection;
+global using Masa.BuildingBlocks.Ddd.Domain.Repositories;
```
Comment (zhenlei520) [inline src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/_Imports.cs:19]: global using sort

**Review by zhenlei520 — CHANGES_REQUESTED** [review]

**Comment by doddgu** [comment]
SonarCloud Failed.

**Comment by wzh425** [comment]
> SonarCloud Failed.
SonarCloud Successful

File: src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    public static async Task<PaginatedList<TEntity>> GetPaginatedListAsync<TEntity>(
+        this IQueryable<TEntity> queryable,
+        Expression<Func<TEntity, bool>> predicate,
+        PaginatedOptions options,
+        CancellationToken cancellationToken = default) where TEntity : class, IEntity
+    {
+        var result = await GetPaginatedListAsync(
+            queryable,
+            predicate,
+            (options.Page - 1) * options.PageSize,
```
Comment (doddgu) [inline src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: I don't think Page needs -1, just PageIndex need it.

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs (hunk 2 above)
Comment (wzh425) [reply src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: The GetPaginatedListAsync method of Masa Framework also uses Page and - 1, which is consistent with it

Comment (doddgu) [reply src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: Fix Masa Framework first.

Comment (wzh425) [reply src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: Page represents which page PageIndex is the index in Pages

Comment (doddgu) [reply src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: Yes.

Comment (doddgu) [reply src/Contrib/Data/Orm/EFCore/Masa.Contrib.Data.EFCore/QueryableExtensions.cs:17]: OK

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**